【hotfix】1.0.0 のアプリが起動しない

## 概要
ユーザー様から「1.0.0 のアプリが起動しない」という連絡が多数来ています。
ビジネス面でかなり問題になってきているので、早急な対応をお願いします。

### 事象報告例
OS: Android 13

[Screen_recording_20231029_151426.webm](https://github.com/tshion/yumemi-inc_android-engineer-codecheck/assets/29895868/4d2d2fbb-3495-40b5-a655-b033c9dfe96e)



## 作業内容
### 完了条件
下記の手順でアプリを起動した時、アプリが正常に起動すること

1. [GitHub Releases](https://github.com/tshion/yumemi-inc_android-engineer-codecheck/releases) の各バージョンに添付されているapk をAndroid 端末にインストールする
1. 前の手順でインストールしたアプリを起動する



### 想定される範囲
* TODO

### 対応しないこと
* TODO



## 進捗メモ
### 1. クラッシュログの確認
Android 13 Emulator で試したところ、下記のクラッシュログを得られた。
その内容から画面遷移周りが怪しいと思われる。

<details>
<summary>※クラッシュログ</summary>

``` log
FATAL EXCEPTION: main
    Process: jp.co.yumemi.android.codecheck, PID: 9345
    java.lang.RuntimeException: Unable to start activity ComponentInfo{jp.co.   yumemi.android.codecheck/jp.co.yumemi.android.code_check.TopActivity}:     java.lang.RuntimeException: Exception inflating jp.co.yumemi.android.   codecheck:navigation/nav_graph line 25
    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3645)
    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782)
    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
    	at android.os.Handler.dispatchMessage(Handler.java:106)
    	at android.os.Looper.loopOnce(Looper.java:201)
    	at android.os.Looper.loop(Looper.java:288)
    	at android.app.ActivityThread.main(ActivityThread.java:7872)
    	at java.lang.reflect.Method.invoke(Native Method)
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
    Caused by: java.lang.RuntimeException: Exception inflating jp.co.yumemi.    android.codecheck:navigation/nav_graph line 25
    	at u0.c0.b(Unknown Source:132)
    	at androidx.navigation.fragment.NavHostFragment.s(SourceFile:527)
    	at androidx.fragment.app.u0.e(SourceFile:66)
    	at androidx.fragment.app.u0.k(SourceFile:151)
    	at androidx.fragment.app.p0.O(SourceFile:63)
    	at androidx.fragment.app.p0.t(SourceFile:36)
    	at androidx.fragment.app.y.onCreate(SourceFile:29)
    	at android.app.Activity.performCreate(Activity.java:8305)
    	at android.app.Activity.performCreate(Activity.java:8284)
    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626)
    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782) 
    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
    	at android.os.Handler.dispatchMessage(Handler.java:106) 
    	at android.os.Looper.loopOnce(Looper.java:201) 
    	at android.os.Looper.loop(Looper.java:288) 
    	at android.app.ActivityThread.main(ActivityThread.java:7872) 
    	at java.lang.reflect.Method.invoke(Native Method) 
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
    Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException:    jp.co.yumemi.android.code_check.item
    	at u0.c0.c(SourceFile:5)
    	at u0.c0.a(SourceFile:93)
    	at u0.c0.a(SourceFile:606)
    	at u0.c0.b(Unknown Source:40)
    	at androidx.navigation.fragment.NavHostFragment.s(SourceFile:527) 
    	at androidx.fragment.app.u0.e(SourceFile:66) 
    	at androidx.fragment.app.u0.k(SourceFile:151) 
    	at androidx.fragment.app.p0.O(SourceFile:63) 
    	at androidx.fragment.app.p0.t(SourceFile:36) 
    	at androidx.fragment.app.y.onCreate(SourceFile:29) 
    	at android.app.Activity.performCreate(Activity.java:8305) 
    	at android.app.Activity.performCreate(Activity.java:8284) 
    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417) 
    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626) 
    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782) 
    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
    	at android.os.Handler.dispatchMessage(Handler.java:106) 
    	at android.os.Looper.loopOnce(Looper.java:201) 
    	at android.os.Looper.loop(Looper.java:288) 
    	at android.app.ActivityThread.main(ActivityThread.java:7872) 
    	at java.lang.reflect.Method.invoke(Native Method) 
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
    Caused by: java.lang.ClassNotFoundException: jp.co.yumemi.android.  code_check.item
    	at java.lang.Class.classForName(Native Method)
    	at java.lang.Class.forName(Class.java:454)
    	at java.lang.Class.forName(Class.java:379)
    	at u0.c0.c(SourceFile:5) 
    	at u0.c0.a(SourceFile:93) 
    	at u0.c0.a(SourceFile:606) 
    	at u0.c0.b(Unknown Source:40) 
    	at androidx.navigation.fragment.NavHostFragment.s(SourceFile:527) 
    	at androidx.fragment.app.u0.e(SourceFile:66) 
    	at androidx.fragment.app.u0.k(SourceFile:151) 
    	at androidx.fragment.app.p0.O(SourceFile:63) 
    	at androidx.fragment.app.p0.t(SourceFile:36) 
    	at androidx.fragment.app.y.onCreate(SourceFile:29) 
    	at android.app.Activity.performCreate(Activity.java:8305) 
    	at android.app.Activity.performCreate(Activity.java:8284) 
    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417) 
    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626) 
    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782) 
    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
    	at android.os.Handler.dispatchMessage(Handler.java:106) 
    	at android.os.Looper.loopOnce(Looper.java:201) 
    	at android.os.Looper.loop(Looper.java:288) 
    	at android.app.ActivityThread.main(ActivityThread.java:7872) 
    	at java.lang.reflect.Method.invoke(Native Method) 
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
    Caused by: java.lang.ClassNotFoundException: jp.co.yumemi.android.  code_check.item
    	at java.lang.Class.classForName(Native Method) 
    	at java.lang.Class.forName(Class.java:454) 
    	at java.lang.Class.forName(Class.java:379) 
    	at u0.c0.c(SourceFile:5) 
    	at u0.c0.a(SourceFile:93) 
    	at u0.c0.a(SourceFile:606) 
    	at u0.c0.b(Unknown Source:40) 
    	at androidx.navigation.fragment.NavHostFragment.s(SourceFile:527) 
    	at androidx.fragment.app.u0.e(SourceFile:66) 
    	at androidx.fragment.app.u0.k(SourceFile:151) 
    	at androidx.fragment.app.p0.O(SourceFile:63) 
    	at androidx.fragment.app.p0.t(SourceFile:36) 
    	at androidx.fragment.app.y.onCreate(SourceFile:29) 
    	at android.app.Activity.performCreate(Activity.java:8305) 
    	at android.app.Activity.performCreate(Activity.java:8284) 
    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417) 
    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626) 
    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782) 
    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101) 
    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307) 
    	at android.os.Handler.dispatchMessage(Handler.java:106) 
    	at android.os.Looper.loopOnce(Looper.java:201) 
    	at android.os.Looper.loop(Looper.java:288) 
    	at android.app.ActivityThread.main(ActivityThread.java:7872) 
    	at java.lang.reflect.Method.invoke(Native Method) 
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```
</details>

### 2. ローカルビルドの結果
* debug ビルドでは問題がなかった
* release ビルドでapk を出力し、それをインストールしたらクラッシュが発生した

上記の結果からビルド構成の違いによるものと考えた。
release ビルドではminify を行っているので、下記の調整をdebug ビルドに加えてみたところ、事象が再現した。

**app/build.gradle**
``` gradle
// 省略
android {
    // 省略
    buildTypes {
        debug {
            debuggable false
            minifyEnabled true
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
        // 省略
    }
    // 省略
```



## 備考
### 参考文献
* TODO

--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -1,21 +1,2 @@
-# Add project specific ProGuard rules here.
-# You can control the set of applied configuration files using the
-# proguardFiles setting in build.gradle.
-#
-# For more details, see
-#   http://developer.android.com/guide/developing/tools/proguard.html
-
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
-
-# Uncomment this to preserve the line number information for
-# debugging stack traces.
-#-keepattributes SourceFile,LineNumberTable
-
-# If you keep the line number information, uncomment this to
-# hide the original source file name.
-#-renamesourcefileattribute SourceFile
+# https://developer.android.com/guide/navigation/use-graph/pass-data#proguard_considerations
+-keepnames class * extends android.os.Parcelable

